docs/spec/currency.rst

```diff
@@ -0,0 +1,515 @@
+.. _currency:
+
+Currency
+========
+
+Overview
+~~~~~~~~
+
+This currency pallet provides an interface for the other pallets to manage balances of different currencies. It is a wrapper around the the `orml-tokens <https://github.com/open-web3-stack/open-runtime-module-library>`_ pallet. As such, accounts have two balances per currency: they have a ``reserved`` amount and a ``free`` amount. Users are able to freely transfer ``free`` balances, but only the parachain pallets are able to operate on ``reserved`` amounts.
```

Comment:
The spec is meant to provide a layer of abstraction, as such I don't think we should mention particular pallets. It will be enough to provide the basic data structures expected of such a module.